test/broker_test/broker_test.go

```diff
@@ -804,6 +812,65 @@ var _ = test.DescribeTestsFor(test.TestCase{
 						})
 					})
 
+					Context("when broker_url is changed but the credentials are missing", func() {
+						var updatedBrokerJSON common.Object
+
+						BeforeEach(func() {
+							updatedBrokerJSON = common.Object{
+								"broker_url": updatedBrokerServer.URL(),
+							}
+						})
+
+						missingCredsTest := func(updatedBrokerJSON common.Object) {
+							ctx.SMWithOAuth.PATCH(web.ServiceBrokersURL+"/"+brokerID).
+								WithJSON(updatedBrokerJSON).
+								Expect().
+								Status(http.StatusBadRequest).JSON().Object().Keys().Contains("error", "description")
+
+							assertInvocationCount(brokerServer.CatalogEndpointRequests, 0)
+
+							ctx.SMWithOAuth.GET(web.ServiceBrokersURL+"/"+brokerID).
+								Expect().
+								Status(http.StatusOK).
+								JSON().Object().
+								ContainsMap(expectedBrokerResponse).
+								Keys().NotContains("services", "credentials")
+						}
```

Comment:
This checks seems unnecessary. With this said the function is unnecessary itself and tests could just make a PATCH call and expect BadRequest.